src/bid/bid.rs

```diff
@@ -60,6 +59,81 @@ impl Borrow<u64> for Bid {
     }
 }
 
+impl PartialEq for Bid {
+    fn eq(&self, other: &Self) -> bool {
+        self.hash().eq(&other.hash())
+    }
+}
+
+// This needs to be between braces since const fn calls passed as const_generics
+// params aren't perfectly supported yet.
+impl Serializable<{ Bid::serialized_size() }> for Bid {
+    type Error = BlindBidError;
+
+    fn from_bytes(buf: &[u8; Self::SIZE]) -> Result<Bid, Self::Error> {
+        let mut one_cipher = [0u8; PoseidonCipher::cipher_size_bytes()];
+        let mut one_scalar = [0u8; 32];
+        let mut one_stealth_address = [0u8; 64];
+        let mut one_u64 = [0u8; 8];
+
+        one_cipher[..]
+            .copy_from_slice(&buf[0..PoseidonCipher::cipher_size_bytes()]);
+        let encrypted_data = PoseidonCipher::from_bytes(&one_cipher)?;
+
+        one_scalar[..]
+            .copy_from_slice(&buf[PoseidonCipher::cipher_size_bytes()..128]);
+        let nonce = BlsScalar::from_bytes(&one_scalar)?;
+
+        one_stealth_address[..].copy_from_slice(&buf[128..192]);
+        let stealth_address = StealthAddress::from_bytes(&one_stealth_address)?;
+
+        one_scalar[..].copy_from_slice(&buf[192..224]);
+        let hashed_secret = BlsScalar::from_bytes(&one_scalar)?;
+
+        one_scalar[..].copy_from_slice(&buf[224..256]);
+        let c = JubJubAffine::from_bytes(&one_scalar)?;
+
+        one_u64[..].copy_from_slice(&buf[256..264]);
+        let eligibility = u64::from_le_bytes(one_u64);
+
+        one_u64[..].copy_from_slice(&buf[264..272]);
+        let expiration = u64::from_le_bytes(one_u64);
+
+        one_u64[..].copy_from_slice(&buf[272..Bid::serialized_size()]);
+        let pos = u64::from_le_bytes(one_u64);
+
+        Ok(Bid {
+            encrypted_data,
+            nonce,
+            stealth_address,
+            hashed_secret,
+            c,
+            eligibility,
+            expiration,
+            pos,
+        })
+    }
+
+    fn to_bytes(&self) -> [u8; Self::SIZE] {
+        let mut buf = [0u8; Self::SIZE];
+        buf[0..PoseidonCipher::cipher_size_bytes()]
```

Comment:
I would use `PoseidonCipher::SIZE` the old methods are still there as aliases, to avoid breaking change.